multi_partner_learning.py

```diff
@@ -425,22 +435,48 @@ def prepare_aggregation_weights(self):
     def aggregate_model_weights(self):
         """Aggregate model weights from the list of models, with a weighted average"""
 
-        weights_per_layer = list(zip(*self.models_weights_list))
-        new_weights = list()
+        if isinstance(self.generate_new_model(), type(LogisticRegression())):
```

Comment:
Do we have to generate a new model here ? Can't we use one of the existing models : self.models_weights_list[0] ?

Reply:
I changed it for 
`if type(self.models_weights_list[0]) is tuple: `
Which is a bit better as keras weights are list